test/controllers/concerns/tag_filter_test.rb

```diff
@@ -0,0 +1,219 @@
+
+    assert output.is_a?(String)
+
+    expected_html =
+      "<option value=\"\">All</option>\n" \
+      "<option value=\"category\">Category</option>\n" \
+      "<option selected=\"selected\" value=\"facility\">Facility</option>\n" \
+      '<option value="partnership">Partnership (Site)</option>'
+
+    assert_equal output, expected_html
+  end
+
+  test '#options_for_per_page' do
+    params = {}
+    filter = TagFilter.new(params)
+    output = filter.options_for_per_page(FakeView.new)
+
+    assert output.is_a?(String)
+
+    expected_html =
+      "<option value=\"10\">10</option>\n" \
+      "<option value=\"20\">20</option>\n" \
+      '<option value="50">50</option>'
+
+    assert_equal output, expected_html
+
+    # with valid value
+    params = { per_page: '20' }
+    filter = TagFilter.new(params)
+    output = filter.options_for_per_page(FakeView.new)
+
+    assert output.is_a?(String)
+
+    expected_html =
+      "<option value=\"10\">10</option>\n" \
+      "<option selected=\"selected\" value=\"20\">20</option>\n" \
+      '<option value="50">50</option>'
+
+    assert_equal output, expected_html
+  end
+
+  test '#next_page_link' do
+    # renders placeholder text when no results found
+    params = {}
+    filter = TagFilter.new(params)
+    scope = filter.with_window(Tag.all)
+
+    output = filter.next_page_link(FakeView.new, scope)
+    assert output.is_a?(String)
+
+    expected_html = 'No more results available'
+    assert_equal output, expected_html
+
+    # now with tags
+    create_list :tag, 25
+
+    params = {}
+    filter = TagFilter.new(params)
+    scope = filter.with_window(Tag.all)
+
+    output = filter.next_page_link(FakeView.new, scope)
+    assert output.is_a?(String)
+
+    expected_html = '<a class="btn btn-link" href="/tags?page_num=2">Next ...</a>'
+    assert_equal output, expected_html
+
+    # second page
+    params = { page_num: '2' }
+    filter = TagFilter.new(params)
+    scope = filter.with_window(Tag.all)
+
+    output = filter.next_page_link(FakeView.new, scope)
+    assert output.is_a?(String)
+
+    expected_html = '<a class="btn btn-link" href="/tags?page_num=3">Next ...</a>'
+    assert_equal output, expected_html
+
+    # last page
+    params = { page_num: '3' }
+    filter = TagFilter.new(params)
+    scope = filter.with_window(Tag.all)
+
+    output = filter.next_page_link(FakeView.new, scope)
+    assert output.is_a?(String)
+
+    expected_html = 'No more results available'
+    assert_equal output, expected_html
+
+    # carries through parameters from rest of filter
+    params = {
+      type: 'facility',
+      name: 'alpha',
+      per_page: '20',
+      page_num: '2'
+    }
+    filter = TagFilter.new(params)
+
+    output = filter.next_page_link(FakeView.new, Tag.all)
+    assert output.is_a?(String)
+
+    expected_html = '<a class="btn btn-link" href="/tags?name=alpha&page_num=3&per_page=20&type=facility">Next ...</a>'
```

Comment:
Can we make this a button and add a "previous" if we've already clicked "next?

Reply:
Maybe? But there is a back button on their browser (hope that doesn't come off as too sarcastic).

When ever I see things like this one sites that do the whole pagination circus with `<< < 0 2 3 ... 45 > >>` type things i think about how i only really use the next page link.

Reply:
> Maybe? But there is a back button on their browser

happy with this

I do think next would benefit from looking like a button though but the PR is still good overall